src/rest.cpp

```diff
@@ -416,34 +423,43 @@ static bool rest_block(const std::any& context,
         pos = pblockindex->GetBlockPos();
     }
 
-    std::vector<std::byte> block_data{};
-    if (!chainman.m_blockman.ReadRawBlock(block_data, pos)) {
-        return RESTERR(req, HTTP_NOT_FOUND, hashStr + " not found");
+    const auto block_data{chainman.m_blockman.ReadRawBlock(pos, block_part)};
+    if (!block_data) {
+        switch (block_data.error()) {
+        case node::ReadRawError::IO: return RESTERR(req, HTTP_INTERNAL_SERVER_ERROR, "I/O error reading " + hashStr);
```

Comment:
nit in the first commit: This is not always an internal server error. It is rarely possible that this is due to a pruning race.

Not sure if anything needs to change, though. Just wanted to leave the note.